src/network/bmobject.py

```diff
@@ -47,39 +71,48 @@ def __init__(self, nonce, expiresTime, objectType, version, streamNumber, data,
         self.inventoryHash = calculateInventoryHash(data)
         # copy to avoid memory issues
         self.data = bytearray(data)
-        self.tag = self.data[payloadOffset:payloadOffset+32]
+        self.tag = self.data[payloadOffset:payloadOffset + 32]
 
     def checkProofOfWorkSufficient(self):
+        """"""
         # Let us check to make sure that the proof of work is sufficient.
         if not protocol.isProofOfWorkSufficient(self.data):
             logger.info('Proof of work is insufficient.')
             raise BMObjectInsufficientPOWError()
 
     def checkEOLSanity(self):
+        """"""
         # EOL sanity check
         if self.expiresTime - int(time.time()) > BMObject.maxTTL:
-            logger.info('This object\'s End of Life time is too far in the future. Ignoring it. Time is %i', self.expiresTime)
-            # TODO: remove from download queue
+            logger.info(
+                'This object\'s End of Life time is too far in the future. Ignoring it. Time is %i',
+                self.expiresTime)
+            # .. todo::  remove from download queue
             raise BMObjectExpiredError()
 
         if self.expiresTime - int(time.time()) < BMObject.minTTL:
-            logger.info('This object\'s End of Life time was too long ago. Ignoring the object. Time is %i', self.expiresTime)
-            # TODO: remove from download queue
+            logger.info(
+                'This object\'s End of Life time was too long ago. Ignoring the object. Time is %i',
+                self.expiresTime)
+            # .. todo::  remove from download queue
             raise BMObjectExpiredError()
 
     def checkStream(self):
+        """"""
         if self.streamNumber not in state.streamsInWhichIAmParticipating:
             logger.debug('The streamNumber %i isn\'t one we are interested in.', self.streamNumber)
             raise BMObjectUnwantedStreamError()
 
     def checkAlreadyHave(self):
```

Comment:
Check if we already have the object (so that we don't duplicate it in inventory or advertise it unnecessarily)